src/common/sockets.cc

```diff
@@ -632,11 +636,15 @@ int32_t tcptowrite(int sock, const void *buff, uint32_t leng, int msecto) {
 	while (sent < leng) {
 		pfd.revents = 0;
 		if (tcppoll(pfd, msecto) < 0) {
+			tcpsetlasterror(TCPETIMEDOUT);
 			return -1;
 		}
 		if (pfd.revents & POLLOUT) {
 			i = tcpsend(sock, ((uint8_t *)buff) + sent, leng - sent, 0);
 			if (i == 0 || (i < 0 && tcpgetlasterror() != TCPEAGAIN)) {
+				if (i == 0) {
+					tcpsetlasterror(TCPETIMEDOUT);
+				}
```

Comment:
If i == 0, then that means while the communication was successful, no data was sent, which may be either correct (as it could not send all the data, so you must resend it again) or a bug (why would we be sending empty data?). Setting ETIMEDOUT feels wrong here as well, as that's not the case really.

Reply:
Same here, decided to not set last tcp error from master.